bats/tests/containers/split-dns-vpn.bats

```diff
@@ -0,0 +1,44 @@
+# bats file_tags=opensuse
+
+load '../helpers/load'
+
+REGISTRY_URL=$(echo "$RD_VPN_TEST_IMAGE" | cut -d'/' -f1)
+
+local_setup() {
+    if ! using_vpn_test_image; then
+        skip "This test requires a connection to the designated VPN."
+    fi
+}
+
+@test 'factory reset' {
+    factory_reset
+}
+
+@test 'start container engine' {
+    start_container_engine
+    wait_for_container_engine
+}
+
+@test 'Can access private registry over VPN from host' {
+    run curl -I -k "https://$REGISTRY_URL/v2"
+    assert_success
+    # We avoid assert_line here due to the trailing carriage return (\r) issues.
+    assert_output --partial "docker-distribution-api-version: registry/2.0"
+}
+
+@test 'Can pull image from private registry over VPN' {
+    run ctrctl pull --quiet "$RD_VPN_TEST_IMAGE"
+    assert_success
+}
+
+@test 'Can verify container access to the registry' {
+    run ctrctl run --rm "$RD_VPN_TEST_IMAGE" curl -I -k "https://$REGISTRY_URL/v2"
```

Comment:
Let's use one of our standard images instead. I just realized that `busybox` doesn't have `curl`, so let's use `nginx`:

```suggestion
    run ctrctl run --rm "$IMAGE_NGINX" curl -I -k "https://$REGISTRY_URL/v2"
```

Unfortunately `nginx` doesn't have `ping`, so we still need to use `busybox` for the next step.

Reply:
I'm okay with changing the image, but don’t you think we're just slowing the test down more and more? We're now pulling three images just to perform a simple test.

Reply:
> don’t you think we're just slowing the test down more and more?

Unfortunately we are. That's why I was hoping to be able to use the same container to run the check on `host.rancher-desktop.internal`, but unfortunately there is no `ping` in the `nginx` image. I can't figure out a way to check routing without `ping` or `traceroute`, and also no known open port on the host.

So having to pull 2 images is unfortunate, but none of our default images seem to have the full set of tools,